doc/tutorial/plugins/libdnf5-plugins.rst

```diff
@@ -0,0 +1,128 @@
+.. _libdnf5 plugins tutorial:
+
+LIBDNF5 Plugins
+===============
+
+These plugins enable changes to the existing dnf5 workflow.
+Modify the library's behavior at defined points, implement new logic,
+or trigger the loading of additional plugins by implementing prepared
+callbacks.
+
+Writing the Plugin
+------------------
+
+Similar to the :ref:`dnf5 plugins tutorial`, we have to implement
+the ``libdnf5::plugin::IPlugin`` interface and override the hooks
+where additional logic is inserted within the dnf5 application lifecycle.
+
+In the following code block, a simple example plugin adds logic at two
+different places: after the ``libdnf5::Base`` object is prepared
+and before the actual transaction starts:
```

Comment:
Also here. I would maybe split the sentence to make it more clear.

"In the following code block, a simple example plugin adds logic in two
different steps.
The first step is after preparing the ``libdnf5::Base`` object.
The second step is before the start of the transaction."